thirdparty/CMakeLists.txt

```diff
@@ -3,76 +3,55 @@ include(FetchContent)
-  add_subdirectory(${spdlog_SOURCE_DIR} ${spdlog_BINARY_DIR} EXCLUDE_FROM_ALL)
-endif()
+CPMAddPackage(
+  NAME spdlog
+  GITHUB_REPOSITORY gabime/spdlog
+  VERSION 1.7.0
+  GIT_SHALLOW TRUE
+  )
 
 ###################################################################################################
 # - thrust/cub ------------------------------------------------------------------------------------
 
 message(STATUS "Fetching Thrust")
 
-FetchContent_Declare(
-    thrust
-    GIT_REPOSITORY https://github.com/thrust/thrust.git
-    # August 28, 2020
-    GIT_TAG        52a8bda46c5c2128414d1d47f546b486ff0be2f0
+CPMAddPackage(
+  NAME thrust
+  GITHUB_REPOSITORY thrust/thrust
+  # August 28, 2020
+  GIT_TAG 52a8bda46c5c2128414d1d47f546b486ff0be2f0
+  VERSION 1.9.10mod
+  DOWNLOAD_ONLY TRUE
 )
 
-FetchContent_GetProperties(thrust)
-if(NOT thrust_POPULATED)
-  FetchContent_Populate(thrust)
-  # We are not using the thrust CMake targets, so no need to call `add_subdirectory()`.
-endif()
 set(THRUST_INCLUDE_DIR "${thrust_SOURCE_DIR}" PARENT_SCOPE)
 
 ###################################################################################################
 # - googletest ------------------------------------------------------------------------------------
 
 if (BUILD_TESTS)
-  message(STATUS "Fetching GoogleTest")
-
-  FetchContent_Declare(
-    googletest
-    GIT_REPOSITORY    https://github.com/google/googletest.git
-    GIT_TAG           release-1.10.0
-    GIT_SHALLOW       true
+  CPMAddPackage(
+    NAME googletest
+    GITHUB_REPOSITORY google/googletest
+    GIT_TAG release-1.10.0
+    VERSION 1.10.0
+    GIT_SHALLOW TRUE
+    OPTIONS
+      "INSTALL_GTEST OFF"
     )
-
-  FetchContent_GetProperties(googletest)
-  if (NOT googletest_POPULATED)
-    FetchContent_Populate(googletest)
-    set(CMAKE_POLICY_DEFAULT_CMP0048 NEW) # suppress warning because googletest doesn't specify its version
-    add_subdirectory(${googletest_SOURCE_DIR} ${googletest_BINARY_DIR} EXCLUDE_FROM_ALL)
-  endif()
 endif()
 
 ###################################################################################################
 # - googlebenchmark -------------------------------------------------------------------------------
 
 if (BUILD_BENCHMARKS)
-  message(STATUS "Fetching GoogleBenchmark")
-
-  FetchContent_Declare(
-    googlebenchmark
-    GIT_REPOSITORY    https://github.com/google/benchmark.git
-    GIT_TAG           v1.5.1
-    GIT_SHALLOW       true
-    )
-
-  FetchContent_GetProperties(googlebenchmark)
-  if (NOT googlebenchmark_POPULATED)
-    FetchContent_Populate(googlebenchmark)
-    set(BENCHMARK_ENABLE_TESTING OFF CACHE INTERNAL "")
-    add_subdirectory(${googlebenchmark_SOURCE_DIR} ${googlebenchmark_BINARY_DIR} EXCLUDE_FROM_ALL)
-  endif()
+  CPMAddPackage(
+    NAME benchmark
+    GITHUB_REPOSITORY google/benchmark
+    VERSION 1.5.2
+    GIT_SHALLOW TRUE
+    OPTIONS
+      "BENCHMARK_ENABLE_TESTING OFF"
+      "BENCHMARK_ENABLE_INSTALL OFF"
+      )
```

Comment:
When I [did this in cuCollections](https://github.com/NVIDIA/cuCollections/blob/035ed1bdabfbe00f662672c16cb0fd5aca9578a2/benchmarks/CMakeLists.txt#L5-L15) I had to workaround for the fact that this was broken with gcc5.4. I guess we dropped support for 5.4 in RAPIDS, so maybe it doesn't matter.

Reply:
Thanks for letting me know. So far it's been working w/o trouble, but it'll be good to know if someone does hit it at some point (though with RMM having dropped support for older CUDA toolkits, hopefully not).